ducktape/mark/resource.py

```diff
@@ -0,0 +1,75 @@
+# Copyright 2016 Confluent Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import copy
+
+from ducktape.mark._mark import Mark
+
+
+class ClusterUseMetadata(Mark):
+    """Provide a hint about how a given test will use the cluster."""
+    def __init__(self, **kwargs):
+        # shallow copy
+        self.metadata = copy.copy(kwargs)
+
+    @property
+    def name(self):
+        return "RESOURCE_HINT_CLUSTER_USE"
+
+    def apply(self, seed_context, context_list):
+        assert len(context_list) > 0, "cluster use annotation is not being applied to any test cases"
+
+        for ctx in context_list:
+            if not ctx.cluster_use_metadata:
+                # only update if non-None and non-empty
+                ctx.cluster_use_metadata = self.metadata
+        return context_list
+
+
+def cluster(**kwargs):
```

Comment:
Sure. The drawback is that there's no validation of the parameters this way, whereas if you explicitly list the parameters you can also get that for free (here we just accept anything, even if its a typo and/or we won't ever use it). I don't feel strongly either way though.